graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/clientapi/ClientApiGenBuilderTest.kt

```diff
@@ -138,4 +139,177 @@ class ClientApiGenBuilderTest {
         assertThat(result2QueryObject.name).isNotNull
         assertThat(result2QueryObject.name).isEqualTo("test")
     }
+
+    @Nested
+    inner class Deprecation {
+
+        @Test
+        fun `adds @Deprecated annotation and reason from schema directives when setting enabled`() {
+            val schema = """
+                type Query {
+                    filter(
+                        nameFilter: String @deprecated(reason: "use idFilter instead"),
+                        idFilter: ID
+                    ): [String]
+                }
+            """.trimIndent()
+
+            val codeGenResult = CodeGen(
+                CodeGenConfig(
+                    schemas = setOf(schema),
+                    packageName = basePackageName,
+                    generateClientApiv2 = true,
+                    maxProjectionDepth = 2,
+                    addDeprecatedAnnotation = true
+                )
+            ).generate()
+
+            assertThat(codeGenResult.javaQueryTypes.size).isEqualTo(1)
+            assertThat(codeGenResult.javaQueryTypes[0].typeSpec.name).isEqualTo("FilterGraphQLQuery")
+            assertThat(codeGenResult.javaQueryTypes[0].typeSpec.typeSpecs).hasSize(1)
+            assertThat(codeGenResult.javaQueryTypes[0].typeSpec.typeSpecs[0].methodSpecs).hasSize(4)
+            assertThat(codeGenResult.javaQueryTypes[0].typeSpec.typeSpecs[0].methodSpecs[1].name).isEqualTo("nameFilter")
+            assertThat(codeGenResult.javaQueryTypes[0].typeSpec.typeSpecs[0].methodSpecs[1].toString()).isEqualTo(
+                """
```

Comment:
I would recommend checking that the generated class has the annotation rather than checking the entire code block to string match against. Same for the other tests.

Reply:
I want to assert that annotation and the javadoc look like expected.
Since both make up the upper part of the code, I rewrote with `startsWith()`. That way one you still easily see the empty that is expected to separate description and `@deprecated` inside the javadoc.

Is this acceptable? Or would you rather me to use `methodSpec`'s `annotations` and `javadoc` for all cases?

Reply:
I guess we can leave it as is in this scenario, makes sense to me.